domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimData.java

```diff
@@ -152,7 +151,13 @@ public Interest getInterest() {
     }
 
     public InterestDate getInterestDate() {
-        return interestDate;
+        if (interestDate != null) {
+            return interestDate;
+        } else if (interest != null) {
+            return interest.getInterestDate();
+        } else {
+            return null;
```

Comment:
Is there an chance that interest date is null? If so shouldn't we use `Optional<InterestDate>`.

Reply:
It's true for the current design but I don't think it's worth doing this refactoring. Once we move the field this will no longer be the case.